en/guides/typesafesessions.md

```diff
@@ -0,0 +1,154 @@
+<div class="titleBlock">
+  <h1>Type-Safe Sessions</h1>
+  <p>How to add Session support to Kitura's Codable Routing</p>
+</div>
+
+## Introduction
+Kitura 2.4 introduces "Type-safe Middleware": middlewares that have a structure and data types that you define, that are instantiated and passed to your handler upon a request. Type-Safe Sessions is an implementation of `TypeSafeMiddleware` that allows an application developer to describe the exact structure and content of a Session, in the form of a `class` conforming to the `TypeSafeSession` protocol.
+
+The key feature is in the name: type safety. The previous implementation of Session that can be used alongside "Raw Routing" provides you with a general `[String: Any]` store, into which any type could be stored and retrieved. This places the burden of type checking on the user: any value retrieved from the session would need a downcast to its expected type, resulting in boilerplate checking and error handling, and the possibility of runtime errors.
+
+In contrast, TypeSafeSession has been designed to provide the following guarantees:
+- If your route is invoked, a session has already been successfully created (or retrieved from a store), and you are passed a (non-optional) instance of your session class. There is no need to check whether a session exists.
+- All data in the session can be accessed directly from your class, with strongly typed properties - there is no need to downcast or handle failure.
```

Comment:
In the point here we have "- there is no need" in the point above we have "There is no need". I prefer the separate sentence but I think we should be consistent about which we use (whichever you choose).